File: sdks/emem-ts/examples/recall.ts
/**
 * Minimal end-to-end smoke example: locate → recall → cite a receipt.
 *
 *   npx tsx examples/recall.ts
 *
 * Or against a local responder:
 *
 *   EMEM_BASE_URL=http://localhost:5051 npx tsx examples/recall.ts
 */

import { Client } from "../src/index.js";

async function main(): Promise<void> {
  const em = new Client();

  const loc = (await em.locate({ place: "Mount Fuji" })) as { cell64?: string };
  const cell = loc.cell64;
  if (!cell) throw new Error(`Mount Fuji did not resolve to a cell64: ${JSON.stringify(loc)}`);
  console.log(`resolved Mount Fuji → cell64 = ${cell}`);

  const facts = (await em.recall({
    cell,
    bands: ["copdem30m.elevation_mean"],
  })) as {
    facts?: Array<{ band?: string; value?: number; cid?: string; tslot?: number }>;
    receipt?: { responder_pubkey?: string; fact_cids?: string[] };
  };

  for (const f of facts.facts ?? []) {
    console.log(
      `  ${f.band ?? "?"} = ${f.value ?? "?"} ` +
        `(cid=${(f.cid ?? "").slice(0, 12)}…, tslot=${f.tslot ?? "?"})`,
    );
  }

  const pk = facts.receipt?.responder_pubkey ?? "";
  const cids = facts.receipt?.fact_cids ?? [];
  console.log(`  receipt: signed by ${pk.slice(0, 16)}… over ${cids.length} fact CID(s)`);
}

main().catch((err) => {
  console.error(err);
  const proc = (globalThis as { process?: { exit: (code: number) => void } }).process;
  proc?.exit(1);
});

File: sdks/emem-py/examples/recall.py
"""Minimal end-to-end smoke example: locate → recall → cite a receipt.

Run against the public hosted instance:

    python examples/recall.py

Or against a local responder:

    EMEM_BASE_URL=http://localhost:5051 python examples/recall.py
"""

from __future__ import annotations

from emem import Client


def main() -> None:
    with Client() as em:
        loc = em.locate("Mount Fuji")
        cell = loc["cell64"]
        print(f"resolved Mount Fuji → cell64 = {cell}")

        facts = em.recall(cell, bands=["copdem30m.elevation_mean"])
        for f in facts.get("facts", []):
            print(
                f"  {f.get('band')} = {f.get('value')} "
                f"(cid={f.get('cid')[:12]}…, tslot={f.get('tslot')})"
            )

        receipt = facts.get("receipt") or {}
        print(
            f"  receipt: signed by {receipt.get('responder_pubkey', '?')[:16]}…"
            f" over {len(receipt.get('fact_cids') or [])} fact CID(s)"
        )


if __name__ == "__main__":
    main()
